components/collector/src/source_collectors/dependency_track/base.py

```diff
@@ -0,0 +1,45 @@
+"""Dependency-Track base collector."""
+
+from base_collectors import SourceCollector
+from collector_utilities.type import URL
+from model import SourceResponses
+
+
+class DependencyTrackBase(SourceCollector):
+    """Dependency-Track base class."""
+
+    PAGE_SIZE = 100  # https://github.com/DependencyTrack/dependency-track/issues/209
+
+    async def _api_url(self) -> URL:
+        """Override to add the API version."""
+        return URL((await super()._api_url()) + "/api/v1")
+
+    def _headers(self) -> dict[str, str]:
+        """Return the headers for the get request."""
+        headers = super()._headers()
+        if api_key := str(self._parameter("private_token")):
+            headers["X-Api-Key"] = api_key
+        return headers
+
+    async def _get_source_responses(self, *urls: URL) -> SourceResponses:
+        """Extend to load multiple pages, if necessary."""
+        responses = SourceResponses()
+        for url in urls:
+            page_nr = 1  # Page numbers start at 1
```

Comment:
Yeah, but that way it's not very clear that the page numbers start at 1 and a future maintainer may think the page number is increased in the wrong location. Since it's surprising it starts at 1 I wanted it to be very clear.